frodo.js

```diff
@@ -3,120 +3,123 @@
  *
```

Comment:
SCRIPT-This function might benefit from using async/await instead of callbacks for better readability.